Tests/DatadogTests/Datadog/Tracing/Utils/ActiveSpanUtilsTest.swift

```diff
@@ -0,0 +1,94 @@
+        XCTAssert(ActiveSpanUtils.getActiveSpan()?.ddContext.spanID == oneSpan.dd.ddContext.spanID)
+
+        oneSpan.finish()
+        XCTAssertNil(ActiveSpanUtils.getActiveSpan())
+    }
+
+    func testsSpanWithoutParentInheritsActiveSpan() throws {
+        let tracer = Tracer.mockAny()
+        let firstSpan = tracer.startSpan(operationName: .mockAny())
+
+        let previousActiveSpan = ActiveSpanUtils.getActiveSpan()
+        let secondSpan = tracer.startSpan(operationName: .mockAny())
+
+        XCTAssertEqual(secondSpan.dd.ddContext.parentSpanID, previousActiveSpan?.dd.ddContext.spanID)
+        XCTAssertEqual(secondSpan.dd.ddContext.spanID,  ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID)
+        XCTAssertEqual(secondSpan.dd.ddContext.parentSpanID, firstSpan.dd.ddContext.spanID)
+
+        secondSpan.finish()
+        XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, firstSpan.dd.ddContext.spanID)
+        firstSpan.finish()
+        XCTAssertNil(ActiveSpanUtils.getActiveSpan())
+    }
+
+    func testsSpanWithParentDoesntInheritActiveSpan() throws {
+        let tracer = Tracer.mockAny()
+        let oneSpan = tracer.startSpan(operationName: .mockAny())
+        let otherSpan = tracer.startSpan(operationName: .mockAny())
+
+        let spanWithParent = tracer.startSpan(operationName: .mockAny(), childOf: oneSpan.context)
+
+        XCTAssertEqual(spanWithParent.dd.ddContext.parentSpanID, oneSpan.dd.ddContext.spanID)
+        spanWithParent.finish()
+        XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, otherSpan.dd.ddContext.spanID)
+        oneSpan.finish()
+        otherSpan.finish()
+    }
+
+    func testActiveSpanIsKeptPerTask() throws {
+        let tracer = Tracer.mockAny()
+        let oneSpan = tracer.startSpan(operationName: .mockAny())
+        let expectation1 = self.expectation(description: "firstSpan created")
+        let expectation2 = self.expectation(description: "secondSpan created")
+
+        DispatchQueue.global(qos: .default).async {
+            let firstSpan = tracer.startSpan(operationName: .mockAny())
+            XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, firstSpan.dd.ddContext.spanID)
+            expectation1.fulfill()
+        }
+
+        DispatchQueue.global(qos: .default).async {
+            Thread.sleep(forTimeInterval: 0.5)
+            XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, oneSpan.dd.ddContext.spanID)
+            let secondSpan = tracer.startSpan(operationName: .mockAny())
+            XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, secondSpan.dd.ddContext.spanID)
+            expectation2.fulfill()
+        }
+
+        XCTAssertEqual(ActiveSpanUtils.getActiveSpan()?.dd.ddContext.spanID, oneSpan.dd.ddContext.spanID)
+        waitForExpectations(timeout: 5, handler: nil)
+        oneSpan.finish()
+    }
+}
```

Comment:
Except unit tests in `ActiveSpanUtilsTests`, how about adding following public api-level test to `TracerTests`?
```swift
func testSendingSpanWithImplicitParent() throws {
    let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
    TracingFeature.instance = .mockWorkingFeatureWith(
        server: server,
        directory: temporaryDirectory
    )
    defer { TracingFeature.instance = nil }

    let tracer = Tracer.initialize(configuration: .init()).dd
    let queue1 = DispatchQueue(label: "\(#function)-queue1")
    let queue2 = DispatchQueue(label: "\(#function)-queue2")

    let rootSpan = tracer.startSpan(operationName: "root operation")

    queue1.sync {
        let childSpan = tracer.startSpan(operationName: "child operation", childOf: rootSpan.context)
        queue2.sync {
            let grandchildSpan = tracer.startSpan(operationName: "grandchild operation", childOf: childSpan.context)
            grandchildSpan.finish()
        }
        childSpan.finish()
    }

    rootSpan.finish()

    let spanMatchers = try server.waitAndReturnSpanMatchers(count: 3)
    let rootMatcher = spanMatchers[2]
    let childMatcher = spanMatchers[1]
    let grandchildMatcher = spanMatchers[0]

    XCTAssertEqual(try rootMatcher.parentSpanID(), "0")
    XCTAssertEqual(try childMatcher.parentSpanID(), try rootMatcher.spanID())
    XCTAssertEqual(try grandchildMatcher.parentSpanID(), try childMatcher.spanID())
}
```
It tests this feature only using public APIs of the `Tracer` + real request captured by intercepting the `URLSession's` protocol (instead of touching internals of `ActiveSpanUtils`). This is a convention we use for all public components (`Tracer`, `Logger`, `RUMMonitor`).

Reply:
Added